.github/workflows/data.yml

```
@@ -0,0 +1,51 @@
+name: Download records & Push to repository
+
+on:
+  push:
+    branches: ["main"]
+  schedule:
+    # Run at 8:00 on working days
+    - cron: 0 8 * * 0-5
+
+
+jobs:
+  download:
+    runs-on: ubuntu-latest
+    if: ${{ !startsWith(github.event.head_commit.message, '[bot]') }}
+    environment:
+      name: download-data
+    env:
+      CI_COMMIT_USER: 'github-actions[bot]'
+      CI_COMMIT_EMAIL: 'github-actions[bot]@users.noreply.github.com'
+      CI_COMMIT_MESSAGE: '[bot] update records'
+      CI_PUSH_REMOTE: git@github.com:$GITHUB_REPOSITORY.git
+      CI_PUSH_BRANCH: 'main'
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
+        persist-credentials: true
+    - uses: actions/setup-python@v4
+      with:
+          python-version: '3.10'
+          cache: 'pip'
```

```suggestion
    - uses: actions/checkout@v4
      with:
        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
        persist-credentials: true
    - uses: actions/setup-python@v5
      with:
          python-version: '3.12'
          cache: 'pip'
```
Updating to latest versions.